src/wasm/literal.cpp

```diff
@@ -403,7 +406,8 @@ std::ostream& operator<<(std::ostream& o, Literal literal) {
       o << "eqref(null)";
       break;
     case Type::i31ref:
-      WASM_UNREACHABLE("TODO: i31ref");
+      o << "i31ref(" << literal.geti31(false) << ")";
```

Comment:
At some point it might be nice to define `Signed` and `Unsigned` as enum values we could use everywhere to represent signedness, but that would be a separate and more widespread change.